hgtv/views/channel.py

```diff
@@ -1,28 +1,41 @@
 # -*- coding: utf-8 -*-
 
 import os
-from flask import render_template, g, flash, jsonify, request
-from coaster.views import load_model, load_models
-from baseframe.forms import render_form, render_redirect
+from flask import render_template, g, jsonify, request, make_response
+from coaster.views import load_model, load_models, render_with
+from baseframe import _
+from baseframe.forms import render_form
 
 from hgtv import app
 from hgtv.views.login import lastuser
 from hgtv.views.video import add_new_video
 from hgtv.forms import ChannelForm, PlaylistForm
 from hgtv.models import Channel, db, Playlist, Video, CHANNEL_TYPE
 from hgtv.uploads import thumbnails, resize_image
+from hgtv.services.channel_details import get_channel_details, get_channel_action_permissions
+from hgtv.services.playlist_details import get_playlist_details
+
+
+def jsonify_channel(data):
+    channel = data['channel']
+    playlist_dict = []
+    videos_count = 'featured'
+    for each_playlist in channel.playlists:
+        playlist_dict.append(get_playlist_details(channel, each_playlist, videos_count))
+    channel_dict = get_channel_details(channel)
+    channel_dict.update(get_channel_action_permissions())
```

Comment:
It's better to get these details from the models themselves, instead of these functions in services module.